packages/components/ToggleThemeButton.tsx

```diff
@@ -1,18 +1,40 @@
 "use client";
 
-import { useColorMode } from "@chakra-ui/react";
+import { useState } from "react";
+
+type Theme = "light" | "dark" | "system";
 
 export default function ToggleThemeButton() {
-  const { toggleColorMode, colorMode } = useColorMode();
+  const [colorMode, setColorMode] = useState(document.documentElement.dataset.theme ?? "system");
+
+  function setNewTheme(newColorMode: Theme) {
+    document.documentElement.dataset.theme = newColorMode;
```

Comment:
is this for `<html data-theme=...` ?

Reply:
Yup

Reply:
It sets the HTML tag first, then the theme cookie, and then we change the react state